BoxBrowseSDK/BoxBrowseSDK/BOXFolderViewController.m

```diff
@@ -379,40 +379,42 @@ - (void)fetchItemsWithCompletion:(void (^)(NSArray *items))completion
             self.title = self.folder.name;
         }
     }];
-
-    // Fetch items
-    [self fetchItemsInFolderID:self.folderID withCompletion:^(NSArray *items, NSError *error) {
-        if (error == nil) {
-            completion(items);
-        } else {
-            self.navigationController.toolbarHidden = YES;
-            [self didFailToLoadItemsWithError:error];
-        }
-        
-        if (self.tableView.visibleCells.count < 1) {
-            [self switchToEmptyStateWithError:error];
-            self.searchController.searchBar.hidden = YES;
-        } else {
-            [self switchToNonEmptyView];
-            self.searchController.searchBar.hidden = NO;
-        }
-    }];
-}
-
-- (void)fetchItemsInFolderID:(NSString *)folderID withCompletion:(void (^)(NSArray *items, NSError *error))completion
-{
-    BOXFolderItemsRequest *request = [self.contentClient folderItemsRequestWithID:folderID];
+    
+    BOXFolderItemsRequest *request = [self.contentClient folderItemsRequestWithID:self.folderID];
     request.SDKIdentifier = BOX_BROWSE_SDK_IDENTIFIER;
     request.SDKVersion = BOX_BROWSE_SDK_VERSION;
     [request setRequestAllItemFields:YES];
     
+    __block NSArray *itemsFromCache = nil;
+    __weak BOXFolderViewController *me = self;
+    
     [request performRequestWithCached:^(NSArray *items, NSError *error) {
-        if (completion) {
-            completion(items, error);
+        itemsFromCache = items;
+        if (completion && error == nil) {
+            completion(items);
+        }
+        
+        if (me.tableView.visibleCells.count < 1) {
+            [me switchToEmptyStateWithError:error];
+            me.searchController.searchBar.hidden = YES;
+        } else {
+            [me switchToNonEmptyView];
+            me.searchController.searchBar.hidden = NO;
         }
     } refreshed:^(NSArray *items, NSError *error) {
-        if (completion) {
-            completion(items, error);
+        if (completion && error == nil) {
+            completion(items);
+        } else if (itemsFromCache == nil) {
```

Comment:
completion is passed down from the parent class when this is invoked so it should always be set, but i'll add a check anyway